tokio/src/sync/mpsc/bounded.rs

```diff
@@ -1009,6 +1066,29 @@ impl<T> fmt::Debug for Sender<T> {
     }
 }
 
+impl<T> Clone for WeakSender<T> {
+    fn clone(&self) -> Self {
+        WeakSender {
+            chan: self.chan.clone(),
+        }
+    }
+}
+
+impl<T> WeakSender<T> {
+    /// Tries to convert a WeakSender into a [`Sender`]. This will return `Some`
+    /// if there are other `Sender` instances alive and the channel wasn't
+    /// previously dropped, otherwise `None` is returned.
+    pub fn upgrade(self) -> Option<Sender<T>> {
+        self.chan.upgrade().map(|tx| Sender::new(tx))
```

Comment:
This triggers clippy:
```suggestion
        self.chan.upgrade().map(Sender::new)
```